packages/astro/src/core/render-context.ts

```diff
@@ -226,23 +258,18 @@ export class RenderContext {
 	#i18nData?: Pick<APIContext, 'currentLocale' | 'preferredLocale' | 'preferredLocaleList'>;
 
 	get i18nData() {
-		if (this.#i18nData) return this.#i18nData;
-		const {
-			pipeline: { i18n },
-			request,
-			routeData,
-		} = this;
-		if (!i18n)
-			return {
-				currentLocale: undefined,
-				preferredLocale: undefined,
-				preferredLocaleList: undefined,
-			};
-		const { defaultLocale, locales, routing } = i18n;
+		if (this.#i18nData) return this.#i18nData
+		const { pipeline: { i18n }, request, routeData, url } = this;
+		if (!i18n) return {
+			currentLocale: undefined,
+			preferredLocale: undefined,
+			preferredLocaleList: undefined
+		}
+		const { defaultLocale, locales, routing } = i18n
 		return (this.#i18nData = {
-			currentLocale: computeCurrentLocale(routeData.route, locales, routing, defaultLocale),
+			currentLocale: computeCurrentLocale(url.pathname, locales, routing, defaultLocale) ?? computeCurrentLocale(routeData.route, locales, routing, defaultLocale),
```

Comment:
We are making two calls to `computeCurrentLocale()`. Previously, these two were made by `createApiContext()` and `createResult()` separately.

I would expect the `pathname` passed to `RenderContext.create()` to be the one true "file-system-matchable" path, but that isn't the case yet - something to tackle with the reroute/rewrite API.

Reply:
The fact that we are making two calls without any inline comment that explains this "anti-pattern" seems like a bug to me.

I know you are explaining this in a comment on Github, but this comment will eventually disappear and the context of this "anti-pattern" won't be explained anymore. Please leave a comment that explains this.

Personally, I would use `routeData.pathname` if it exists, and `routeData.route` otherwise. `Request#url` isn't reliable because in SSG has this value: `/en.html` and we won't be able to pick the current locale anymore.